packages/combo-box/test/basic.test.js

```diff
@@ -271,6 +271,39 @@ describe('Properties', () => {
 
         expect(spy.calledOnce).to.be.true;
       });
+
+      it('should not fire twice when the custom value set listener causes blur', () => {
+        const spy = sinon.spy();
+        comboBox.addEventListener('custom-value-set', spy);
+
+        // Emulate opening the overlay that causes blur
+        comboBox.addEventListener('custom-value-set', () => {
+          comboBox.blur();
+        });
+
+        comboBox.open();
+        input.value = 'foo';
+        input.dispatchEvent(new CustomEvent('input'));
+        comboBox.close();
+
+        expect(spy.calledOnce).to.be.true;
+      });
+
+      it('should fire twice when another custom value is committed by the user', () => {
+        const spy = sinon.spy();
+        comboBox.addEventListener('custom-value-set', spy);
+
+        comboBox.open();
+        input.value = 'foo';
+        input.dispatchEvent(new CustomEvent('input'));
+        comboBox.close();
+
+        input.value = 'bar';
+        input.dispatchEvent(new CustomEvent('input'));
+        focusout(input);
+
+        expect(spy.calledTwice).to.be.true;
+      });
```

Comment:
Add a test for https://github.com/vaadin/web-components/pull/3478/files#diff-bf0d744fa330e331944495a3b4fc16a99bc9f89d618011f89be659cb56b8a734R711

```suggestion
      });
      
      it('should fire when setting the same custom value after clearing', () => {
        const spy = sinon.spy();
        comboBox.addEventListener('custom-value-set', spy);

        input.value = 'foo';
        input.dispatchEvent(new CustomEvent('input'));
        focusout(input);

        input.value = '';
        input.dispatchEvent(new CustomEvent('input'));
        focusout(input);

        spy.resetHistory();
        input.value = 'foo';
        input.dispatchEvent(new CustomEvent('input'));
        focusout(input);

        expect(spy.calledOnce).to.be.true;
      });
```